### PR TITLE
Move dialogues to v02 of the toolkit

### PR DESCRIPTION
**Pull request by justinvdm**
We hope to have solved the issue we get with `EndState`s for non-session-based channels with praekelt/vumi-jssandbox-toolkit#164, moving dialogues over to v02 would be a good test for this.

### CODE REVIEW

**Comment by justinvdm** [comment]
Ready for review.

File: go/apps/dialogue/tests/test_vumi_app.js [hunk 1]
```diff
@@ -1,204 +1,192 @@
...
-            user: {current_state: "choice-1"},
-            helper_metadata: {delivery_class: 'ussd'},
-            content: null,
-            next_state: "choice-1",
-            response: (
-                "^What is your favourite colour\\?[^]" +
-                "1. Red[^]" +
-                "2. Blue"
-            )
+require('mocha-as-promised')();
```
Comment (hodgestar) [inline go/apps/dialogue/tests/test_vumi_app.js:1]: Do we still need this? Perhaps we can just update our Mocha?

**Comment by hodgestar** [comment]
Other than the question about Mocha, looks awesome. :)

File: go/apps/dialogue/tests/test_vumi_app.js (hunk 1 above)
Comment (justinvdm) [inline go/apps/dialogue/tests/test_vumi_app.js:1]: I think we still need it unfortunately :/ we use grunt-mocha-test, which uses mocha ~1.12.0.

Comment (hodgestar) [inline go/apps/dialogue/tests/test_vumi_app.js:1]: Ah.

**Comment by hodgestar** [comment]
:+1:

**Pull request merged**